src/plugins/saved_objects_management/public/management_section/register_section.tsx

```diff
@@ -0,0 +1,216 @@
+import { DataPublicPluginStart } from '../../../data/public';
+import { StartDependencies, SavedObjectsManagementPluginStart } from '../plugin';
+import {
+  ISavedObjectsManagementServiceRegistry,
+  SavedObjectsManagementActionServiceStart,
+} from '../services';
+import { SavedObjectsTable } from './objects_table';
+import { SavedObjectEdition } from './object_view';
+import { getAllowedTypes } from './../lib';
+
+interface RegisterOptions {
+  core: CoreSetup<StartDependencies, SavedObjectsManagementPluginStart>;
+  sections: ManagementSetup['sections'];
+  serviceRegistry: ISavedObjectsManagementServiceRegistry;
+}
+
+const title = i18n.translate('savedObjectsManagement.managementSectionLabel', {
+  defaultMessage: 'Saved Objects',
+});
+
+export const registerManagementSection = ({ core, sections, serviceRegistry }: RegisterOptions) => {
+  const kibanaSection = sections.getSection('kibana');
+  if (!kibanaSection) {
+    throw new Error('`kibana` management section not found.');
+  }
+  kibanaSection.registerApp({
+    id: 'objects',
+    title,
+    order: 10,
+    mount: async ({ element, basePath, setBreadcrumbs }) => {
+      const [coreStart, { data }, pluginStart] = await core.getStartServices();
+      const allowedTypes = await getAllowedTypes(coreStart.http);
+      const capabilities = coreStart.application.capabilities;
+
+      ReactDOM.render(
+        <I18nProvider>
+          <HashRouter basename={basePath}>
+            <Switch>
+              <Route path={'/:service/:id'} exact={true}>
+                <RedirectToHomeIfUnauthorized capabilities={capabilities}>
+                  <SavedObjectsEditionPage
+                    coreStart={coreStart}
+                    serviceRegistry={serviceRegistry}
+                    setBreadcrumbs={setBreadcrumbs}
+                  />
+                </RedirectToHomeIfUnauthorized>
+              </Route>
+              <Route path={'/'} exact={false}>
+                <RedirectToHomeIfUnauthorized capabilities={capabilities}>
+                  <SavedObjectsTablePage
+                    coreStart={coreStart}
+                    dataStart={data}
+                    serviceRegistry={serviceRegistry}
+                    actionRegistry={pluginStart.actions}
+                    allowedTypes={allowedTypes}
+                    setBreadcrumbs={setBreadcrumbs}
+                  />
+                </RedirectToHomeIfUnauthorized>
+              </Route>
+            </Switch>
+          </HashRouter>
+        </I18nProvider>,
+        element
+      );
+
+      return () => {
+        ReactDOM.unmountComponentAtNode(element);
+      };
+    },
+  });
+};
+
+const RedirectToHomeIfUnauthorized: React.FunctionComponent<{
+  capabilities: Capabilities;
+}> = ({ children, capabilities }) => {
+  const allowed = capabilities?.management?.kibana?.objects ?? false;
+  if (!allowed) {
+    window.location.hash = '/home';
+    return null;
+  }
+  return children! as React.ReactElement;
+};
```

Comment:
Two things on that `RedirectToHomeIfUnauthorized`:

1/ This was previously done using the `requireUICapability` property when registering the angular route. I did not find any equivalent in `management` NP API, so I had to code it @elastic/kibana-app-arch can you confirm there is no API/utils to do that?

2/ Regarding `window.location.hash = '/home';`, I can't use `react-router-dom`'s `Redirect` here (or any related API), as it would append the basePath (see `<HashRouter basename={basePath}>`) and redirect to `/app/kibana#/management/objects/home` instead. I could use NP's `navigateToApp('kibana', { path: '#/home'})`, but that would trigger a full refresh, so I kept that. I'm open to any better alternative. Btw, we got FTR tests covering this.